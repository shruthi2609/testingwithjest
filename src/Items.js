import React from 'react';
import PropTypes from 'prop-types';
function Items(props) {
  const { items = [] } = props;

  // A single item in the list, render a span.
  if (items.length === 1) {
    return <div>{items[0]}</div>;
  }

  // Multiple items on the list, render a list.
  if (items.length > 1) {
    return (
      <ul>
        {items.map(item => <li>{item}</li>)}
      </ul>
    );
  }

  // No items on the list, render an empty message.
  return <div>No items in list</div>;
}

Items.propTypes = {
  items: PropTypes.array,
};

Items.defaultProps = {
  items: []
};

export default Items;
