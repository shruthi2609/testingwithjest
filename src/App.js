import React, { Component } from 'react';
import Items from './Items';

class App extends Component {
  render() {
    const items = [
      "React",
      "js",
      "express"
      
    ];
    return (
      <Items items={items} />
    );
  }
}
export default App;

