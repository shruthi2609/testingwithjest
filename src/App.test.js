// Counter App Test file
/*import { render,screen } from "@testing-library/react"
import UserEvent from "@testing-library/user-event"
import App from "./App"
test("increment action",()=>{
    render(<App></App>)
    const counter=screen.getByTestId("counter")
    expect(counter.textContent).toBe("0")
    const incrementbutton=screen.getByText("+")
    UserEvent.click(incrementbutton)
    UserEvent.click(incrementbutton)
    expect(counter.textContent).toBe("2")
})
test("decrement action",()=>{
    render(<App></App>)
    const counter=screen.getByTestId("counter")
    const decrementbutton=screen.getByText("-")
    UserEvent.click(decrementbutton)
    UserEvent.click(decrementbutton)
    expect(counter.textContent).toBe("-2")
})
test("combine incr and decr",()=>{
    render(<App></App>)
    const counter=screen.getByTestId("counter")
    const incrementbutton=screen.getByText("+")
    const decrementbutton=screen.getByText("-")
    UserEvent.click(incrementbutton)
    UserEvent.click(incrementbutton)
    UserEvent.click(decrementbutton)
    UserEvent.click(decrementbutton)
    UserEvent.click(decrementbutton)
    expect(counter.textContent).toBe("-1")

})
//regular expressions
import { render,screen } from "@testing-library/react"
import App from "./App"
test("check app is rendering a text value",()=>{
    render(<App></App>)
    expect(screen.getByText(/React/)).toBeInTheDocument()
})*/
//snapshot testing
import  TestRenderer  from "react-test-renderer";
import Items from "./Items";
it("no props",()=>{
    const tree=TestRenderer .create(<Items></Items>)
    expect(tree).toMatchSnapshot();
})
it("render one prop",()=>{
    const tech=["React"]
    const tree= TestRenderer .create(<Items items={tech}></Items>)
    expect(tree).toMatchSnapshot()
})
it("render multiple props",()=>{
    const tech=["React","js","express"]
    const tree= TestRenderer .create(<Items items={tech}></Items>)
    expect(tree).toMatchSnapshot()
})

