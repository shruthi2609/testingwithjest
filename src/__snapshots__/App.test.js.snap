// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no props 1`] = `
<div>
  No items in list
</div>
`;

exports[`render multiple props 1`] = `
<ul>
  <li>
    React
  </li>
  <li>
    js
  </li>
  <li>
    express
  </li>
</ul>
`;

exports[`render one prop 1`] = `
<div>
  React
</div>
`;
